r2/r2/controllers/front.py

```diff
@@ -955,7 +944,7 @@ def GET_search_reddits(self, query, reverse, after, count, num):
               recent=VMenu('t', TimeMenu, remember=False),
               restrict_sr=VBoolean('restrict_sr', default=False),
               include_facets=VBoolean('include_facets', default=False),
-              syntax=VOneOf('syntax', options=SearchQuery.known_syntaxes))
+              syntax=VOneOf('syntax', options=g.search.SearchQuery.known_syntaxes))
```

Comment:
Here's another tricky case of reading something of `g` at import time.

I'm testing locally a separate validator for this that defers reading the choices off of `g` until runtime:

```python
class VSearchSyntax(VOneOf):
    def run(self, val):
        # Deferred setting of self.options to avoid reading from pylons.g
        # at import time (which is when Validators are instantiated)
        self.options = g.search.SearchQuery.known_syntaxes
        return VOneOf.run(self, val)
```